third_party/terraform/resources/resource_sql_database_instance.go.erb

```diff
@@ -1006,6 +1032,12 @@ func resourceSqlDatabaseInstanceDelete(d *schema.ResourceData, meta interface{})
 		return err
 	}
 
+	// Check if deletion protection is enabled.
+ 
+	if d.Get("settings.0.deletion_protection").(bool) {
```

Comment:
I'm pretty sure this is best achieved with lifecycle_params: prevent_destroy - I don't think it belongs here.

Reply:
There is precedent for doing this instead of using lifecycle params. https://github.com/GoogleCloudPlatform/magic-modules/pull/3450